src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
```

Comment:
Critically, it's not a default value. It's an EMPTY value. A default value is, say, 5. If nothing is specified you get a 5. This is bad for all kinds of reasons, not the least of which that you cannot distinguish default 5 from explicit 5 - or if you can it's because you are managing additional out-of-band information.

Think of the empty value as a type-specific None, because that's exactly what it is. Rather than carrying around Some(foo) and None until the end of time, you carry foo and the in-type instance of foo - which YOU choose - which means None.

Requiring metadata keys to supply an empty means the fundamental signature of Map is `K => V` instead of `K => Option[V]` which eliminates a pervasive and frankly disastrous layer of wrapping.